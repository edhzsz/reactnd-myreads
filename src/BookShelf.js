import React from 'react'
import BooksGrid from './BooksGrid.js'

const BookShelf = ({ title, books }) => (
  <div className="bookshelf">
    <h2 className="bookshelf-title">{title}</h2>
    <div className="bookshelf-books">
      <BooksGrid books={books} />
    </div>
  </div>
  );

export default BookShelf;