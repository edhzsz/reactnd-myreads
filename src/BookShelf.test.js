import React from 'react';
import ReactDOM from 'react-dom';
import BookShelf from './BookShelf.js';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Component: BookShelf', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BookShelf />, div);
  });

  it('should match its empty snapshot', () => {
    const tree = toJson(shallow(
      <BookShelf />
     ));

    expect(tree).toMatchSnapshot();
  });

  it('should render the props', () => {
    const books = [
      {
        title: "To Kill a Mockingbird",
        authors: ["Harper Lee"],
        imageLinks: {
          smallThumbnail:"http://books.google.com/books/content?id=PGR2AwAAQBAJ&printsec=frontcover&img=1&zoom=5&imgtk=AFLRE73-GnPVEyb7MOCxDzOYF1PTQRuf6nCss9LMNOSWBpxBrz8Pm2_mFtWMMg_Y1dx92HT7cUoQBeSWjs3oEztBVhUeDFQX6-tWlWz1-feexS0mlJPjotcwFqAg6hBYDXuK_bkyHD-y&source=gbs_api",
          thumbnail:"http://books.google.com/books/content?id=PGR2AwAAQBAJ&printsec=frontcover&img=1&zoom=1&imgtk=AFLRE73-GnPVEyb7MOCxDzOYF1PTQRuf6nCss9LMNOSWBpxBrz8Pm2_mFtWMMg_Y1dx92HT7cUoQBeSWjs3oEztBVhUeDFQX6-tWlWz1-feexS0mlJPjotcwFqAg6hBYDXuK_bkyHD-y&source=gbs_api",
        },
        shelf: "currentlyReading",
        id: "nggnmAEACAAJ"
      },
      {
        title: "Ender's Game",
        authors: ["Orson Scott Card"],
        imageLinks: {
          smallThumbnail:"http://books.google.com/books/content?id=yDtCuFHXbAYC&printsec=frontcover&img=5&zoom=1&imgtk=AFLRE72RRiTR6U5OUg3IY_LpHTL2NztVWAuZYNFE8dUuC0VlYabeyegLzpAnDPeWxE6RHi0C2ehrR9Gv20LH2dtjpbcUcs8YnH5VCCAH0Y2ICaKOTvrZTCObQbsfp4UbDqQyGISCZfGN&source=gbs_api",
          thumbnail:"http://books.google.com/books/content?id=yDtCuFHXbAYC&printsec=frontcover&img=1&zoom=1&imgtk=AFLRE72RRiTR6U5OUg3IY_LpHTL2NztVWAuZYNFE8dUuC0VlYabeyegLzpAnDPeWxE6RHi0C2ehrR9Gv20LH2dtjpbcUcs8YnH5VCCAH0Y2ICaKOTvrZTCObQbsfp4UbDqQyGISCZfGN&source=gbs_api",
        },
        shelf: "currentlyReading",
        id: "sJf1vQAACAAJ"
      }
    ];

    const tree = toJson(shallow(
      <BookShelf title="test shelf" books={books} />
     ));

    expect(tree).toMatchSnapshot();
  });
});