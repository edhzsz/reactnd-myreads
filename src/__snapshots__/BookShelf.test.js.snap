// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: BookShelf should match its empty snapshot 1`] = `
<div
  className="bookshelf"
>
  <h2
    className="bookshelf-title"
  />
  <div
    className="bookshelf-books"
  >
    <BooksGrid />
  </div>
</div>
`;

exports[`Component: BookShelf should render the props 1`] = `
<div
  className="bookshelf"
>
  <h2
    className="bookshelf-title"
  >
    test shelf
  </h2>
  <div
    className="bookshelf-books"
  >
    <BooksGrid
      books={
        Array [
          Object {
            "authors": Array [
              "Harper Lee",
            ],
            "id": "nggnmAEACAAJ",
            "imageLinks": Object {
              "smallThumbnail": "http://books.google.com/books/content?id=PGR2AwAAQBAJ&printsec=frontcover&img=1&zoom=5&imgtk=AFLRE73-GnPVEyb7MOCxDzOYF1PTQRuf6nCss9LMNOSWBpxBrz8Pm2_mFtWMMg_Y1dx92HT7cUoQBeSWjs3oEztBVhUeDFQX6-tWlWz1-feexS0mlJPjotcwFqAg6hBYDXuK_bkyHD-y&source=gbs_api",
              "thumbnail": "http://books.google.com/books/content?id=PGR2AwAAQBAJ&printsec=frontcover&img=1&zoom=1&imgtk=AFLRE73-GnPVEyb7MOCxDzOYF1PTQRuf6nCss9LMNOSWBpxBrz8Pm2_mFtWMMg_Y1dx92HT7cUoQBeSWjs3oEztBVhUeDFQX6-tWlWz1-feexS0mlJPjotcwFqAg6hBYDXuK_bkyHD-y&source=gbs_api",
            },
            "shelf": "currentlyReading",
            "title": "To Kill a Mockingbird",
          },
          Object {
            "authors": Array [
              "Orson Scott Card",
            ],
            "id": "sJf1vQAACAAJ",
            "imageLinks": Object {
              "smallThumbnail": "http://books.google.com/books/content?id=yDtCuFHXbAYC&printsec=frontcover&img=5&zoom=1&imgtk=AFLRE72RRiTR6U5OUg3IY_LpHTL2NztVWAuZYNFE8dUuC0VlYabeyegLzpAnDPeWxE6RHi0C2ehrR9Gv20LH2dtjpbcUcs8YnH5VCCAH0Y2ICaKOTvrZTCObQbsfp4UbDqQyGISCZfGN&source=gbs_api",
              "thumbnail": "http://books.google.com/books/content?id=yDtCuFHXbAYC&printsec=frontcover&img=1&zoom=1&imgtk=AFLRE72RRiTR6U5OUg3IY_LpHTL2NztVWAuZYNFE8dUuC0VlYabeyegLzpAnDPeWxE6RHi0C2ehrR9Gv20LH2dtjpbcUcs8YnH5VCCAH0Y2ICaKOTvrZTCObQbsfp4UbDqQyGISCZfGN&source=gbs_api",
            },
            "shelf": "currentlyReading",
            "title": "Ender's Game",
          },
        ]
      }
    />
  </div>
</div>
`;
